File: BleModule/src/main/java/com/topdon/lms/sdk/utils/SPUtils.java
package com.topdon.lms.sdk.utils;

import android.content.Context;

/**
 * Stub implementation for SPUtils
 * 
 * @author Copilot
 */
public class SPUtils {
    private static SPUtils instance;
    
    public static SPUtils getInstance(Context context) {
        if (instance == null) {
            instance = new SPUtils();
        }
        return instance;
    }
    
    public Object get(String key, Object defaultValue) {
        // Stub implementation - return default value
        return defaultValue;
    }
}

File: android/libapp/src/main/java/com/topdon/lms/utils/SPUtils.kt
package com.topdon.lms.utils

/**
 * Stub implementation of SPUtils for compilation
 */
class SPUtils private constructor() {
    companion object {
        @Volatile
        private var instance: SPUtils? = null
        
        @JvmStatic
        fun getInstance(): SPUtils {
            return instance ?: synchronized(this) {
                instance ?: SPUtils().also { instance = it }
            }
        }
    }
    
    fun put(key: String, value: String) {}
    
    fun put(key: String, value: Int) {}
    
    fun put(key: String, value: Boolean) {}
    
    fun getString(key: String): String = ""
    
    fun getString(key: String, defaultValue: String): String = defaultValue
    
    fun getInt(key: String): Int = 0
    
    fun getInt(key: String, defaultValue: Int): Int = defaultValue
    
    fun getBoolean(key: String): Boolean = false
    
    fun getBoolean(key: String, defaultValue: Boolean): Boolean = defaultValue
}